File: 02_ProgrammingFundamentals/_05_Lists/_01_Lecture/_05_ListManipulationAdvanced.java
package _02_ProgrammingFundamentals._05_Lists._01_Lecture;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _05_ListManipulationAdvanced {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<Integer> numbersList = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
        String console = scan.nextLine();

        // Manipulate the list based on the given commands
        while (!console.equals("end")) {
            String[] commandLine = console.split("\\s+");

            if (commandLine[0].equals("Contains")) {
                if (numbersList.contains(Integer.parseInt(commandLine[1]))) {
                    System.out.println("Yes");
                } else {
                    System.out.println("No such number");
                }
            } else if (commandLine[0].equals("Print")) {
                boolean isChanged = false;

                if (commandLine[1].equals("even")) {
                    for (int i = 0; i < numbersList.size(); i++) {
                        if (numbersList.get(i) % 2 == 0) {
                            isChanged = true;
                        }
                    }
                    
                    if (isChanged) {
                        numbersList.stream().forEach(e -> {
                            if (e % 2 == 0) {
                                System.out.print(e + " ");
                            }
                        });

                        System.out.println();
                    }
                } else if (commandLine[1].equals("odd")) {
                    for (int i = 0; i < numbersList.size(); i++) {
                        if (numbersList.get(i) % 2 != 0) {
                            isChanged = true;
                        }
                    }

                    if (isChanged) {
                        numbersList.stream().forEach(e -> {
                            if (e % 2 != 0) {
                                System.out.print(e + " ");
                            }
                        });

                        System.out.println();
                    }
                }
            } else if (commandLine[0].equals("Get")) {
                int sum = 0;

                for (int i = 0; i < numbersList.size(); i++) {
                    sum += numbersList.get(i);
                }

                System.out.println(sum);
            } else if (commandLine[0].equals("Filter")) {
                numbersList.stream().forEach(e -> {
                    if (commandLine[1].equals("<")) {
                        if (e < Integer.parseInt(commandLine[2])) {
                            System.out.print(e + " ");
                        }
                    } else if (commandLine[1].equals(">")) {
                        if (e > Integer.parseInt(commandLine[2])) {
                            System.out.print(e + " ");
                        }
                    } else if (commandLine[1].equals(">=")) {
                        if (e >= Integer.parseInt(commandLine[2])) {
                            System.out.print(e + " ");
                        }
                    } else if (commandLine[1].equals("<=")) {
                        if (e <= Integer.parseInt(commandLine[2])) {
                            System.out.print(e + " ");
                        }
                    }
                });

                System.out.println();
            }

            console = scan.nextLine();
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_05_Lists/_01_Lecture/_05_ListManipulationAdvanced/_05_ListManipulationAdvanced.cs
using System;
using System.Linq;

public class _05_ListManipulationAdvanced
{
    public static void Main(string[] args)
    {
        // Read input values
        List<int> numbersList = Console.ReadLine().Split().Select(int.Parse).ToList();
        string console = Console.ReadLine();

        // Add variable
        bool isChanged = false;

        // Manipulate the list based on the given commands
        while (!console.Equals("end"))
        {
            string[] commandLine = console.Split();
            string command = commandLine[0].ToLower();

            if (command.Equals("add"))
            {
                numbersList.Add(int.Parse(commandLine[1]));
                isChanged = true;
            }
            else if (command.Equals("remove"))
            {
                numbersList.Remove(int.Parse(commandLine[1]));
                isChanged = true;
            }
            else if (command.Equals("removeat"))
            {
                numbersList.RemoveAt(int.Parse(commandLine[1]));
                isChanged = true;
            }
            else if (command.Equals("insert"))
            {
                numbersList.Insert(int.Parse(commandLine[2]), int.Parse(commandLine[1]));
                isChanged = true;
            }
            else if (command == "contains")
            {
                Console.WriteLine(numbersList.Contains(int.Parse(commandLine[1])) ? "Yes" : "No such number");
            }
            else if (command == "printeven")
            {
                Console.WriteLine(string.Join(' ', numbersList.Where(n => n % 2 == 0)));
            }
            else if (command == "printodd")
            {
                Console.WriteLine(string.Join(' ', numbersList.Where(n => n % 2 != 0)));
            }
            else if (command == "getsum")
            {
                Console.WriteLine(numbersList.Sum());
            }
            else if (command == "filter")
            {
                if (commandLine[1] == "<")
                {
                    Console.WriteLine(string.Join(' ', numbersList.Where(n => n < int.Parse(commandLine[2]))));
                }
                else if (commandLine[1] == ">")
                {
                    Console.WriteLine(string.Join(' ', numbersList.Where(n => n > int.Parse(commandLine[2]))));
                }
                else if (commandLine[1] == ">=")
                {
                    Console.WriteLine(string.Join(' ', numbersList.Where(n => n >= int.Parse(commandLine[2]))));
                }
                else if (commandLine[1] == "<=")
                {
                    Console.WriteLine(string.Join(' ', numbersList.Where(n => n <= int.Parse(commandLine[2]))));
                }
            }

            console = Console.ReadLine();
        }

        if (isChanged)
        {
            Console.WriteLine(string.Join(' ', numbersList));
        }
    }
}
